File: homeWork_18_05_2023/java/com/example/homework_18_05_23/MainActivity.java
package com.example.homework_18_05_23;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements OnItemClick{

    ArrayList<TelephoneModel> arrayListNames=new ArrayList<>();

    TelephoneAdapter telephoneAdapter;

    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        telephoneAdapter=new TelephoneAdapter(arrayListNames, this);
        recyclerView.findViewById(R.id.recyclerView);
        recyclerView.setAdapter(telephoneAdapter);

        arrayListNames.add(new TelephoneModel(R.drawable.tel_one, "Iphone 11", "PRO MAX", "2020", "128GB", "48000 СОМ"));
        arrayListNames.add(new TelephoneModel(R.drawable.tel_two, "SAMSUNG S20", "GALAXY", "2018", "256GB", "240000 COM"));
    }

    @Override
    public void onItemClick(TelephoneModel telephoneModel) {
        Intent intent=new Intent(this, DetailActivity.class);
        intent.putExtra("telephone", telephoneModel);
        startActivity(intent);
    }
}

File: finalTestThreeMonthGeeks/main/java/com/example/finalhomework/MainActivity.kt
package com.example.finalhomework

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.example.finalhomework.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    lateinit var categoryAdapter: CategoryAdapter
    lateinit var categoryRecyclerView: RecyclerView
    lateinit var binding:ActivityMainBinding

    lateinit var foodAdapter: FoodAdapter
    lateinit var foodRecyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding= ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setAdapters()

    }
    private fun setCategory():ArrayList<Category>{
        return arrayListOf(
            Category(R.drawable.i340, "Takeaways"),
            Category(R.drawable.i341, "Grocery"),
            Category(R.drawable.i342, "Convenience"),
            Category(R.drawable.i343, "Pharmacy")
        )
    }

    private fun setFoods():ArrayList<Food>{
        return arrayListOf(
            Food(R.drawable.i322, "Burger Craze", "OPEN", R.drawable.i719, "1.5km away", R.drawable.i721, 4.6,"American Burgers")
        )
    }

    private fun setAdapters(){
        categoryAdapter= CategoryAdapter(setCategory())
        categoryRecyclerView.adapter=categoryAdapter
        categoryRecyclerView=binding.recyclerViewCategory

        foodAdapter= FoodAdapter(setFoods())
        foodRecyclerView=binding.recyclerViewFoods
        foodRecyclerView.adapter=foodAdapter
    }
}